portlayer/attach/connector.go

```diff
@@ -137,22 +137,30 @@ func (c *Connector) Remove(id string) {
 
 // takes the base connection, determines the ID of the source and stashes it in the map
 func (c *Connector) processIncoming(conn net.Conn) {
+	var err error
+	defer func() {
+		if err != nil && conn != nil {
```

Comment:
So if `err == nil`, we don't close `conn`? This seems to be a change from before where `conn.Close()` was called unconditionally.